pkg/virt-handler/cmd-client/client.go

```diff
@@ -628,3 +629,27 @@ func (c *VirtLauncherClient) GetFilesystems() (v1.VirtualMachineInstanceFileSyst
 
 	return filesystemList, nil
 }
+
+// Exec the command with args on the guest and return the resulting status code, stdOut and error
+func (c *VirtLauncherClient) Exec(domainName, command string, args []string) (int, string, error) {
+	request := &cmdv1.ExecRequest{
+		DomainName: domainName,
+		Command:    command,
+		Args:       args,
+	}
+	exitCode := -1
+	stdOut := ""
+
+	ctx, cancel := context.WithTimeout(context.Background(), shortTimeout)
+	defer cancel()
+
```

Comment:
using the hard coded `shortTimeout` here makes an assumption the the exec probe is something that will return within a small timeframe `5 seconds`

That's probably the case the majority of the time, but I think it's technically possible for someone to configure an exec probe with a custom timeout which might exceed this pre-defined 5 second timeout. In that case, we'd always fail.

since the timeout is part of the api, we should probably extend it to be accounted for during the execution

Reply:
Right, I just read through the note here again: https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/#configure-probes
We should support the timeout specified on the probe and do the same.
It may still result in resource leakage if they user command does not terminate properly, but at least we relieve K8s.
I'll have a look.

Reply:
Right now we also only wait for the qemu-exec command to return in 10s.
This should probably use the same value?
https://github.com/kubevirt/kubevirt/pull/5038/files#diff-a70c73881228b460a83e4bbfa48cffe8c55af0ba91fea5f48dabd3390ffdea5bR67